Source/WebCore/platform/mediastream/gstreamer/GStreamerMediaStreamSource.cpp

```diff
@@ -629,11 +630,15 @@ bool webkitMediaStreamSrcSetStream(WebKitMediaStreamSrc* self, MediaStreamPrivat
 
 static void webkitMediaStreamSrcPushVideoSample(WebKitMediaStreamSrc* self, GstSample* gstsample)
 {
+   if (self->firstVideoSampleSeen == FALSE)
+      self->firstVideoSampleSeen = TRUE;
     self->videoSrc.pushSample(gstsample);
 }
 
 static void webkitMediaStreamSrcPushAudioSample(WebKitMediaStreamSrc* self, GstSample* gstsample)
 {
+    if (self->videoSrc.src()  && self->firstVideoSampleSeen == FALSE)
```

Comment:
Is this fix specific for broadcom? I'm not sure if this is something we always want to block. Do you have a testcase we can try?

Reply:
This is not specific to broadcom.  We see audio delay , if we push audio sample before video samples.

Reply:
This is not specific to broadcom. We see audio delay , if first video samples are pushed before first audio sample.

Reply:
Hi @eocanha, in libwebrtc case, webrtc engine is responsible for the synchronization. The web app may configure media stream to present audio and video as soon as possible, effectively disabling a/v sync. I think that is why sync on the clock was disabled by default for media stream pipeline (https://bugs.webkit.org/show_bug.cgi?id=190677). And it is one of the reasons why on Westeros sink side we also have an immediate output mode that disables platform a/v synchronization if any is present.

Unfortunately, that is not enough. In some cases, we see that webrtc audio source starts providing data much earlier than video source (sometimes diff is ~2 seconds). audio data just accumulates in the pipeline, until video is provided. this is due to streamsynchronizer in playbin, that blocks pads until both streams has some data. This is causing a significant delay between audio and video. 

This PR is just a workaround. Idea is that it will drop audio until the first video frame is enqueued as well.

Reply:
Ok, now I understand and can see how it makes sense. Still, as half of the code seems specific to Broadcom elements and upstream is now using gstwebrtc instead of libwebrtc, I'm going to approve this PR here but I'm not going to try to port it upstream.